python/kotekan/visbuffer.py

```diff
@@ -449,36 +512,111 @@ def from_buffer(cls, buffer, size_frame, num_time, num_freq):
             size_frame, num_elements, num_prod, num_ev, align_valid=True
         )
 
-        # TODO: Python 3 - use native_str for compatibility
         raw = buffer.view(dtype=frame_struct)
         data = raw["data"]
         metadata = raw["metadata"]
         valid_frames = raw["valid"]
 
         ctime = metadata["ctime"]
         fpga_seq = metadata["fpga_seq"]
+
         num_prod = metadata["num_prod"]
 
         time = np.ndarray(
             shape=(num_time, num_freq),
-            dtype=[
-                (native_str("fpga_count"), np.uint64),
-                (native_str("ctime"), time_spec),
-            ],
+            dtype=[("fpga_count", np.uint64), ("ctime", np.float64),],
         )
+
+        # flatten time index map (we only need one value per time slot, but we have one per
+        # frame/frequency)
         for t in range(num_time):
```

Comment:
Ideally this would be a vectorized numpy version, but again I guess time is too short to fix this one.